### ISSUE TITLE
Mounting union includes the root filesystem

### ISSUE THREAD

**Issue by robertbaker** [rclone/rclone#5139]
#### What is the problem you are having with rclone?
I updated to the latest beta and now my union includes the entire root filesystem and it's not mounting the remote.

This only seems to affect mounting a union, mounting a remote directly works as expected.

rclone is including the root fileystem in the union, even though it's not specified as a upstream branch. This issue only affects the recent 1.55 beta's and I believe were caused by the refactoring of the fsPath and splitPath code, see commit https://github.com/rclone/rclone/commit/f17d7c00126ed567e0fbf2710f2726cf92c26dfe

Reverting to 1.5.4.1 does not have this bug and works as expected..

#### What is your rclone version (output from `rclone version`)
```
v1.55.0-beta.5314.e59acd16c
- os/arch: linux/amd64
- go version: go1.15.8
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Ubuntu 20.04

####  Which cloud storage system are you using? (e.g. Google Drive)
n/a

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
`rclone mount gdrive-union:/ /data/union`

`ls /data/union` -> shows `ls / && ls /data/local`

```
[gdrive]
type = drive
(omitted for brevity)

[gdrive-local]
type = local

[gdrive-union]
type = union
upstreams = gdrive-local:/data/local gdrive:/:nc
create_policy = ff
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
n/a

**Comment by darthShadow** [comment]
Paging @ncw

**Comment by darthShadow** [comment]
Just to verify the root cause, please try with this build too: https://beta.rclone.org/v1.55.0-beta.5274.ff6868900/

**Comment by ncw** [comment]
I've had a go at fixing this here - can you give it a try?

[v1.55.0-beta.5327.507e8a188.fix-5139-union-init](https://beta.rclone.org/branch/fix-5139-union-init/v1.55.0-beta.5327.507e8a188.fix-5139-union-init/) on branch [fix-5139-union-init](https://github.com/rclone/rclone/tree/fix-5139-union-init) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.55
